### PR TITLE
✨ Support dead embeds

### PR DESCRIPTION
**Pull request by Enriqe**
Dead embeds will behave like regular embeds but won't be interactive (i.e. they won't trigger the tooltip). 

This PR makes it so that publishers will have to explicitly opt-in to the interactive mode. They will be able to do this using the `interactive="true"` attribute on the embed.

Follow ups will be sent for validation / visual tests.

### CODE REVIEW

File: extensions/amp-story/1.0/amp-story-embedded-component.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ export const EXPANDABLE_COMPONENTS = {
     '0 78.62 23"/></g></svg>',
     actionIcon: ActionIcon.EXPAND,
     localizedStringId: LocalizedStringId.AMP_STORY_TOOLTIP_EXPAND_TWEET,
-    selector: 'amp-twitter',
+    selector: 'amp-twitter[interactive=yes]',
```
Comment (newmuis) [inline extensions/amp-story/1.0/amp-story-embedded-component.js]: I think we would want this to equally apply to all embeds.  Is there a way that we don't have to repeat this every time we add a new embed?

File: extensions/amp-story/1.0/amp-story-page.js [hunk 2]
```diff
@@ -471,13 +476,35 @@ export class AmpStoryPage extends AMP.BaseElement {
...
   findAndPrepareEmbeddedComponents_(forceResize = false) {
-    scopedQuerySelectorAll(this.element, EMBEDDED_COMPONENTS_SELECTORS)
+    this.addClickShieldToEmbeds_();
+    this.resizeInteractiveEmbeds_(forceResize);
+  }
+
+  /**
+   * Adds a pseudo element on top of the embed to block clicks from going into
+   * the iframe.
+   */
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js]: Nit: here and below, `@private`

File: extensions/amp-story/1.0/amp-story-page.js [hunk 3]
```diff
@@ -471,13 +476,35 @@ export class AmpStoryPage extends AMP.BaseElement {
...
+  addClickShieldToEmbeds_() {
+    scopedQuerySelectorAll(this.element, EMBEDDED_COMPONENTS_SELECTORS).forEach(
+        el => {
+          this.resources_.mutateElement(el, () => {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js]: Calling `mutateElement` has a bit of overhead in the runtime. Could we query the elements, and then call mutateElement only once?

File: extensions/amp-story/1.0/amp-story-page.js [hunk 4]
```diff
@@ -1001,7 +1028,7 @@ export class AmpStoryPage extends AMP.BaseElement {
    * @private
    */
   toggleLoadingSpinner_(isActive) {
-    this.getVsync().mutate(() => {
+    this.resources_.mutateElement(this.element, () => {
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js]: `this.mutateElement(() => {...})`

File: extensions/amp-story/1.0/amp-story-page.js [hunk 5]
```diff
@@ -471,13 +476,35 @@ export class AmpStoryPage extends AMP.BaseElement {
...
-   * Finds embedded components in page and prepares them for their expanded view
-   * animation.
+   * Finds embedded components in page and prepares them.
    * @param {boolean=} forceResize
    * @private
    */
   findAndPrepareEmbeddedComponents_(forceResize = false) {
-    scopedQuerySelectorAll(this.element, EMBEDDED_COMPONENTS_SELECTORS)
+    this.addClickShieldToEmbeds_();
+    this.resizeInteractiveEmbeds_(forceResize);
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js]: Naming nit: `embed` could be anything really (any content within the page), could we use EmbeddedComponents again?

File: extensions/amp-story/1.0/amp-story-embedded-component.js (hunk 1 above)
Comment (gmajoulet) [reply extensions/amp-story/1.0/amp-story-embedded-component.js]: I imagine we could dynamically append `[interactive=true]` at the end of the selectors automatically, but at this point I think having it here is easier (still one line), whereas one more layer of looping through and editing the config when there already are many really scares me :(

File: extensions/amp-story/1.0/amp-story-embedded-component.js [hunk 6]
```diff
@@ -88,6 +88,12 @@ const LAUNCHABLE_COMPONENTS = {
   },
 };
 
+/** @const {!Array<string>} */
+const INTERACTIVE_ATTR_NAMES = [
+  '[interactive="true"]',
+  '[interactive=""]',
+  '[interactive]'];
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-embedded-component.js]: I find this much more complicated than what we had before, we replaced one line with multiple lines of configuration and loops and string manipulations :(
Also, `amp-twitter[interactive]:not([interactive=false])` might be better :)

Jon's call!

File: extensions/amp-story/1.0/amp-story-page.js [hunk 7]
```diff
@@ -471,13 +476,40 @@ export class AmpStoryPage extends AMP.BaseElement {
...
+
+  /**
+   * Adds a pseudo element on top of the embed to block clicks from going into
+   * the iframe.
+   * @private
+   */
+  addClickShieldToEmbeds_() {
+    const els = [];
+    scopedQuerySelectorAll(this.element, EMBEDDED_COMPONENTS_SELECTORS).forEach(
+        el => els.push(el));
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js]: Do you need to create this extra array, or can you just loop through the result of `scopedQuerySelectorAll()`?

File: extensions/amp-story/1.0/amp-story-embedded-component.js (hunk 6 above)
Comment (Enriqe) [reply extensions/amp-story/1.0/amp-story-embedded-component.js]: Revised it. LMK what you think.

File: extensions/amp-story/1.0/amp-story-page.js [hunk 8]
```diff
@@ -471,13 +476,37 @@ export class AmpStoryPage extends AMP.BaseElement {
...
+  }
+
+  /**
+   * Adds a pseudo element on top of the embed to block clicks from going into
+   * the iframe.
+   * @private
+   */
+  addClickShieldToEmbeds_() {
+    this.mutateElement(() => {
+      scopedQuerySelectorAll(this.element, EMBEDDED_COMPONENTS_SELECTORS)
```
Comment (gmajoulet) [inline extensions/amp-story/1.0/amp-story-page.js]: Sorry to be picky about this, but can we query the DOM before the `mutateElement`, and add the `...embeddedComponents_` suffix to the method name to be consistent?

I was thinking something like
```
const componentEls = scopedQuerySelectorAll();
this.mutateElement(() => {
  componentEls.forEach(...);
});
```

Doesn't matter so much here, but it's a good practice to optimize the JS execution order :)

**Comment by Enriqe** [comment]
PTAL @newmuis

**Pull request merged**